src/apps/common/submodules/appSelector/appSelector.js

```diff
@@ -0,0 +1,377 @@
+define(function(require) {
+	var $ = require('jquery'),
+		_ = require('lodash'),
+		isotope = require('isotope'),
+		monster = require('monster');
+
+	var appSelector = {
+		// Defines API requests not included in the SDK
+		requests: {},
+
+		// Define the events available for other apps
+		subscribe: {
+			'common.appSelector.render': 'appSelectorRender',
+			'common.appSelector.renderPopup': 'appSelectorRenderPopup',
+			'common.appSelector.getSelectedApps': 'appSelectorGetSelectedApps'
+		},
+
+		appFlags: {
+			appSelector: {
+				tagFilters: [
+					'all',
+					'reseller',
+					'carrier',
+					'developer',
+					'end_user',
+					'reporting',
+					'integration',
+					'admin',
+					'system_admin'
+				]
+			}
+		},
+
+		/**
+		 * Store getter
+		 * @param  {Array|String} [path]
+		 * @param  {*} [defaultValue]
+		 * @return {*}
+		 */
+		appSelectorGetStore: function(path, defaultValue) {
+			var self = this,
+				store = ['_store', 'appSelector'];
+			return _.get(
+				self,
+				_.isUndefined(path)
+					? store
+					: _.flatten([store, _.isString(path) ? path.split('.') : path]),
+				defaultValue
+			);
+		},
+
+		/**
+		 * Store setter
+		 * @param  {Array|String|*} path|value
+		 * @param  {*} [value]
+		 */
+		appSelectorSetStore: function(path, value) {
+			var self = this,
+				hasValue = _.toArray(arguments).length === 2,
+				store = ['_store', 'appSelector'];
+			_.set(
+				self,
+				hasValue
+					? _.flatten([store, _.isString(path) ? path.split('.') : path])
+					: store,
+				hasValue ? value : path
+			);
+		},
+
+		/**
+		 * Render app selector
+		 * @param  {Object} args
+		 * @param  {jQuery} args.container  Element that will contain the app selector
+		 * @param  {('all'|'account'|'user')} [args.scope='all'] App list scope
+		 * @param  {Boolean} [args.forceFetch=false]  Force to fetch app data from API instead of using the cached one
+		 * @param  {String[]} [args.selectedAppIds=[]]  Pre-selected application IDs
+		 * @param  {Function} [args.callback]  Optional callback to be executed after render
+		 */
+		appSelectorRender: function(args) {
+			var self = this,
+				scope = _.get(args, 'scope', 'all'),
+				forceFetch = _.get(args, 'forceFetch', false),
+				selectedAppIds = _.get(args, 'selectedAppIds', []),
+				$container = args.container,
+				initTemplate = function initTemplate(apps) {
+					var selectedApps = self.appSelectorGetStore('selectedApps', []),
+						tagCount = _.chain(apps).flatMap('tags').countBy().value(),
+						filters = _.transform(self.appFlags.appSelector.tagFilters, function(filters, filterName) {
+							filters[filterName] = (filterName === 'all') ? _.size(apps) : _.get(tagCount, filterName, 0);
+						}, {}),
+						dataTemplate = {
+							apps: apps,
```

Comment:
Looks like this is an object as well? An array would be better, especially for sorting purposes.